pkg/config/config.go

```diff
@@ -116,7 +121,11 @@ func (c *Config) Set(keys []string, value string) {
 		}
 		m = entry
 	}
-	m.SetEntry(keys[len(keys)-1], yamlmap.StringValue(value))
+	val := yamlmap.StringValue(value)
+	if value == "" {
+		val = yamlmap.NullValue()
+	}
+	m.SetEntry(keys[len(keys)-1], val)
```

Comment:
Initially, I assumed we'd have the ability to set the config to null values, not that all empty strings will be null values.

I don't know a definitive way this will be a problem assuming this is YAML and should hopefully work 🤞

Reply:
It is YAML so they are more or less equivalent, which is why I went this direction instead of introducing a new function like `SetNull`. It felt mostly like implementation detail that the user shouldn't be concerned with. The user still receives an empty string using `Get` if they `Set` an empty string which felt like the only contract that we need to maintain here.